### PR TITLE
feat: migration rename files

### PR DESCRIPTION
**Pull request by ImCesar**
- Closes [Unify-516](https://cypress-io.atlassian.net/browse/UNIFY-516)

### Additional details
business logic to move specs and provide useful data to GQL 

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/data-context/test/unit/config-generator.spec.ts [hunk 1]
```diff
@@ -1,75 +1,100 @@
 import snapshot from 'snap-shot-it'
-import { createConfigString } from '../../src/util/migration'
+import { createConfigString, getSpecs } from '../../src/util/migration'
+import { expect } from 'chai'
+const util = require('util')
+const exec = util.promisify(require('child_process').exec)
```
Comment (lmiller1990) [inline packages/data-context/test/unit/config-generator.spec.ts:5]: Generally you will use the `execa` package, we already have it, it does all the async stuff for you so you can just do `await execa(...)`. Grep the code base for examples.

File: packages/data-context/test/unit/config-generator.spec.ts [hunk 2]
```diff
@@ -1,75 +1,100 @@
...
+      snapshot(generatedConfig)
+    })
+  })
 
-    snapshot(generatedConfig)
+  describe('spec renaming', () => {
+    it('should rename all specs', async () => {
+      const { stdout } = await exec('git rev-parse --show-toplevel')
+      const repoRoot = stdout.trim()
+      const componentDirPath = `${repoRoot}/system-tests/projects/migration/cypress/component`
```
Comment (lmiller1990) [inline packages/data-context/test/unit/config-generator.spec.ts:85]: Might be good to `path.normalize` this, just a good practice.

File: packages/data-context/src/util/migration.ts [hunk 3]
```diff
@@ -78,3 +137,72 @@ function createTestingTypeTemplate (testingType: 'e2e' | 'component', pluginPath
...
+      to: renameSpecPath(specPath),
+    }
+  })
+
+  specs.forEach((spec) => {
+    fs.moveSync(spec.from, spec.to)
+  })
+}
+
+async function getSpecFiles (dirPath: string) {
```
Comment (lmiller1990) [inline packages/data-context/src/util/migration.ts:194]: FYI this is basically just re-inventing the `globby` library @ImCesar. Just do

```js
await globby('**/*/your-glob', { filesOnly: true })
```

It's free!

File: packages/launchpad/src/migration/fragments/HighlightedFile.vue [hunk 4]
```diff
@@ -1,51 +1,29 @@
...
+  parts: readonly FilePart[]
   highlightClass: string
 }>()
 
-const fileParts = computed(() => {
-  let result:RegExpExecArray | null
-  let start:number = 0
-  const allParts: { text:string, highlight:boolean }[] = []
-
-  while ((result = props.highlightRegExp.exec(props.file))) {
```
Comment (lmiller1990) [inline packages/launchpad/src/migration/fragments/HighlightedFile.vue:27]: @elevatebart nuked this, better just to use the built-in feature of regular expressions called "named groups". EG

```
`cypress\/(?<dir>${defaultFolder})\/.*?(?<ext>[._-]?[s|S]pec.|[.])(?=[j|t]s[x]?)`
```

```
(?<name-of-group>)
You can name the group and do the replace w/o while/slice/index hacks. Whole thing is just this now: https://github.com/cypress-io/cypress/pull/19807/files#diff-0075775a5a7e9fabfcc8e083c6d37c56439b7055d9a0ec04af8459cd2f68e6deR10-R26

File: packages/data-context/src/util/migration.ts (hunk 3 above)
Comment (ImCesar) [reply packages/data-context/src/util/migration.ts:193]: After messing around with the e2e test I was wondering if this is the path we want to go. Do we only want files for this? If they have a nested directory like we have in our test project with `someDir/someFile` do we want to rename those as well? In the UI it shows them being renamed but we haven't renamed them

File: packages/launchpad/cypress/e2e/migration.cy.ts [hunk 5]
```diff
@@ -54,7 +90,7 @@ describe('Migration', () => {
       cy.findByText(defaultMessages.migration.wizard.step4.button).click()
       cy.findByText(defaultMessages.migration.wizard.step5.button).click()
 
-      cy.findByText(defaultMessages.setupWizard.selectFramework.description).should('be.visible')
+      // cy.findByText(defaultMessages.setupWizard.selectFramework.description).should('be.visible')
```
Comment (ImCesar) [inline packages/launchpad/cypress/e2e/migration.cy.ts:93]: @elevatebart is this working right now? It does nothing when clicked, also I think we agreed it shouldn't go to that page

Comment (elevatebart) [reply packages/launchpad/cypress/e2e/migration.cy.ts:93]: For now, it should still go to this page. If it was broken, that is a bug.

Comment (elevatebart) [reply packages/launchpad/cypress/e2e/migration.cy.ts:93]: let me see what's happening

**Comment by lmiller1990** [comment]
@ImCesar @elevatebart Got this working (including the component "life update" migration step.

This needs a LOT of polish, going to merge it now since I got it green, we can make a new PR for more work.

**Pull request merged**